pandas/io/excel/_xlsxwriter.py

```diff
@@ -222,11 +227,9 @@ def write_cells(
         # Write the frame cells using xlsxwriter.
         sheet_name = self._get_sheet_name(sheet_name)
 
-        if sheet_name in self.sheets:
-            wks = self.sheets[sheet_name]
-        else:
+        wks = self.book.get_worksheet_by_name(sheet_name)
```

Comment:
i would change `.book` to be a private attribute `._book` and then expose the `.book` property (followups for sure)

Reply:
Would the property be something besides just `return self._book`? In other words, why a property?